### PR TITLE
enh: support kind check condition in python script

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Fixes: https://github.com/lfortran/lfortran/issues/4399, https://github.com/lfortran/lfortran/issues/3574
Towards: #4017
Ref: https://github.com/lfortran/lfortran/pull/3573#discussion_r1509374634

### CODE REVIEW

File: src/libasr/intrinsic_func_registry_util_gen.py [hunk 1]
```diff
@@ -866,6 +884,10 @@ def add_verify_arg_type_src(func_name):
         compute_arg_types(3 * indent, no_of_args, "x.m_args")
         condition, cond_in_msg = compute_arg_condition(no_of_args, args_lists)
         src += 3 * indent + f'ASRUtils::require_impl({condition}, "Unexpected args, {func_name} expects {cond_in_msg} as arguments", x.base.base.loc, diagnostics);\n'
+        if same_kind_arg:
+            compute_arg_kinds(3 * indent, no_of_args)
+            condition = compute_kind_condition(no_of_args)
+            src += 3 * indent + f'ASRUtils::require_impl({condition}, "Kind of all the arguments of `{func_name}` must be same", x.base.base.loc, diagnostics);\n'
```
Comment (certik) [inline src/libasr/intrinsic_func_registry_util_gen.py:890]: Maybe (not sure):
```suggestion
            src += 3 * indent + f'ASRUtils::require_impl({condition}, "Kind of all the arguments of `{func_name}` must be the same", x.base.base.loc, diagnostics);\n'
```

**Comment by certik** [comment]
I think that's fine. Tests have to pass.

**Comment by HarshitaKalani** [comment]
I'm not sure what is causing the error. I think it might be visible in mac, can someone help me on that?

File: tests/reference/asr-merge_bits_comp-23a3e67.stderr [hunk 2]
```diff
@@ -1,5 +1,5 @@
-semantic error: The second argument of 'merge_bits' intrinsic must be the same type and kind as first argument
+ASR verify pass error: Kind of all the arguments of Mergebits must be the same
```
Comment (certik) [inline tests/reference/asr-merge_bits_comp-23a3e67.stderr:1]: This needs to fail in AST->ASR as a semantic error

**Review by certik — APPROVED** [review]

**Comment by Pranavchiku** [comment]
Just a small suggestion whenever we update test references, we shall prefer to delete all previous references and then update it.

```
rm -rf ./tests/reference/*
./run_tests.py -u
```

These commands shall do the job. The reason to do is, when you delete a file and / or rename it ( for example in this PR you renamed `tests/error/intrinsics_02` to `tests/error/intrinsics2` ) if you do not remove the references, the old ones will stay ( refer [pull/4462/files](https://github.com/lfortran/lfortran/pull/4462/files/b41beb4ad0755680a9546c58eb48fee8bf749abd#diff-f2ce0cee4a210bb671eb83b0def9935db323ed77b4f979acf796bb08b310bc6c) ). These do not harm us any way as the example is already deleted, but a good practice to first remove and then update.

**Comment by Pranavchiku** [comment]
While searching for which commit added this file, I came across https://stackoverflow.com/questions/19875104/how-to-find-out-in-which-commit-a-particular-code-was-added, it has a bunch of cool git tricks that I encountered for the first time. Worth giving it a read.

**Comment by HarshitaKalani** [comment]
Oh I see, you're right. I'll make sure to follow that

**Pull request merged**